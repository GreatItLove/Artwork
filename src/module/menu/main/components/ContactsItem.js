import React from 'react';
import { ListItem } from 'material-ui/List';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

import { CONTACTS_PATH } from '../../../../config/routesPatch';

const propTypes = {
  style: PropTypes.object.isRequired
};

/**
 * @function ContactsItem
 * @param {Object} style
 * @returns {XML}
 */
function ContactsItem({ style }) {
  return (
    <ListItem
      key={CONTACTS_PATH}
      primaryText="Contacts"
      innerDivStyle={style}
      leftIcon={<i className="material-icons">contact_mail</i>}
      containerElement={<NavLink to={CONTACTS_PATH} exact />}
    />
  );
}

ContactsItem.propTypes = propTypes;

export default ContactsItem;
