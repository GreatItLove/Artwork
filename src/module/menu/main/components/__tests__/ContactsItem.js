import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import ContactsItem from '../ContactsItem';

describe('ContactsItem', () => {
  it('Snapshot of ContactsItem', () => {
    const wrapper = shallow(<ContactsItem style={{}} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
