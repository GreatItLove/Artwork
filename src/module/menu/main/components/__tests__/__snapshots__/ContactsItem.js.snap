// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsItem Snapshot of ContactsItem 1`] = `
<ListItem
  autoGenerateNestedIndicator={true}
  containerElement={
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      exact={true}
      to="/contacts"
    />
  }
  disableKeyboardFocus={false}
  disabled={false}
  initiallyOpen={false}
  innerDivStyle={Object {}}
  insetChildren={false}
  leftIcon={
    <i
      className="material-icons"
    >
      contact_mail
    </i>
  }
  nestedItems={Array []}
  nestedLevel={0}
  onKeyboardFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onNestedListToggle={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  open={null}
  primaryText="Contacts"
  primaryTogglesNestedList={false}
  secondaryTextLines={1}
/>
`;
